nav2_dwb_controller/dwb_core/src/dwb_local_planner.cpp

```diff
@@ -32,89 +32,95 @@
 #include "dwb_core/illegal_trajectory_tracker.hpp"
+#include "dwb_msgs/msg/critic_score.hpp"
+#include "nav_2d_msgs/msg/twist2_d.hpp"
 #include "nav_2d_utils/conversions.hpp"
 #include "nav_2d_utils/parameters.hpp"
 #include "nav_2d_utils/tf_help.hpp"
-#include "nav_2d_msgs/msg/twist2_d.hpp"
-#include "dwb_msgs/msg/critic_score.hpp"
+#include "nav2_lifecycle/lifecycle_node.hpp"
 #include "pluginlib/class_list_macros.hpp"
-#include "dwb_core/exceptions.hpp"
 
 namespace dwb_core
 {
 
-void loadBackwardsCompatibleParameters(const std::shared_ptr<rclcpp::Node> & nh)
-{
-  std::vector<std::string> critic_names;
-  RCLCPP_INFO(nh->get_logger(),
-    "DWBLocalPlanner", "No critics configured! Using the default set.");
-  critic_names.push_back("RotateToGoal");       // discards trajectories that move forward when
-                                                //   already at goal
-  critic_names.push_back("Oscillation");        // discards oscillating motions (assisgns cost -1)
-  critic_names.push_back("ObstacleFootprint");  // discards trajectories that move into obstacles
-  critic_names.push_back("GoalAlign");          // prefers trajectories that make the
-                                                //   nose go towards (local) nose goal
-  critic_names.push_back("PathAlign");          // prefers trajectories that keep the
-                                                //   robot nose on nose path
-  critic_names.push_back("PathDist");           // prefers trajectories on global path
-  critic_names.push_back("GoalDist");           // prefers trajectories that go towards
-                                                //   (local) goal, based on wave propagation
-  nh->set_parameters({rclcpp::Parameter("critics", critic_names)});
-  /* *INDENT-OFF* */
-  nav_2d_utils::moveParameter(nh, "path_distance_bias", "PathAlign.scale", 32.0, false);
-  nav_2d_utils::moveParameter(nh, "goal_distance_bias", "GoalAlign.scale", 24.0, false);
-  nav_2d_utils::moveParameter(nh, "path_distance_bias", "PathDist.scale", 32.0);
-  nav_2d_utils::moveParameter(nh, "goal_distance_bias", "GoalDist.scale", 24.0);
-  nav_2d_utils::moveParameter(nh, "occdist_scale",      "ObstacleFootprint.scale", 0.01);
-
-  nav_2d_utils::moveParameter(nh, "max_scaling_factor", "ObstacleFootprint.max_scaling_factor", 0.2);  // NOLINT
-  nav_2d_utils::moveParameter(nh, "scaling_speed",      "ObstacleFootprint.scaling_speed", 0.25);
-  /* *INDENT-ON* */
-}
-
-DWBLocalPlanner::DWBLocalPlanner()
-: traj_gen_loader_("dwb_core", "dwb_core::TrajectoryGenerator"),
+DWBLocalPlanner::DWBLocalPlanner(
+  nav2_lifecycle::LifecycleNode::SharedPtr node, TFBufferPtr tf,
+  CostmapROSPtr costmap_ros)
+: node_(node),
+  tf_(tf),
+  costmap_ros_(costmap_ros),
+  traj_gen_loader_("dwb_core", "dwb_core::TrajectoryGenerator"),
   goal_checker_loader_("dwb_core", "dwb_core::GoalChecker"),
   critic_loader_("dwb_core", "dwb_core::TrajectoryCritic")
 {
 }
 
-void DWBLocalPlanner::initialize(
```

Comment:
if removing `initialize`,  we should remove definition in header

Reply:
True, removed.